### PR TITLE
expose scan_cache table generation to python

### PR DESCRIPTION
**Pull request by rsxdalv**
Allows for the functionality to be used with servers/gradio projects:

```python
from huggingface_hub import scan_cache_dir
from huggingface_hub.commands.scan_cache import get_table

hf_cache_info = scan_cache_dir()

table = get_table(0, hf_cache_info)
print(table)
```

### CODE REVIEW

File: src/huggingface_hub/commands/scan_cache.py
Comment (Wauplin) [inline src/huggingface_hub/commands/scan_cache.py:86]: Slight preference to have `cache_info` first and then `verbosity` as a keyword-only argument. 

```suggestion
        print(get_table(hf_cache_info, verbosity=self.verbosity))


def get_table(hf_cache_info: HFCacheInfo, *, verbosity: int = 0) -> str:
```

Means than you should use it like this `table = get_table(hf_cache_info, verbosity=0)` or even `table = get_table(hf_cache_info)` (more readable / easy to use in my opinion).

**Review by Wauplin — COMMENTED** [review]
Thanks @rsxdalv, makes sense to me. The changes looks good to me. Left a minor comment regarding the helper signature.

In addition to this, could you also add 

```
### scan_cache.get_table

[[autodoc]] huggingface_hub.commands.scan_cache.get_table
```

to the [cache.md package reference.](https://github.com/huggingface/huggingface_hub/blob/9a9b8c137dbe27e553f85e737a14289479a86641/docs/source/en/package_reference/cache.md?plain=1#L21). This would add it to the official documentation under https://huggingface.co/docs/huggingface_hub/package_reference/cache. You must add a quick docstring to `get_table` to explain what it does, its inputs and an example. You can take inspiration from [this docstring](https://github.com/huggingface/huggingface_hub/blame/c4b59c4d95c96c590f21f909e81dc44d7a05c84f/src/huggingface_hub/utils/_cache_manager.py#L500) for example. Thanks a lot in advance!

**Comment by rsxdalv** [comment]
Thanks for the information! I added a docstring and the reference to en/package_reference/cache.md

I was not sure about the ko/package_reference/cache.md , so I did not add it there.

File: src/huggingface_hub/commands/scan_cache.py
Comment (Wauplin) [inline src/huggingface_hub/commands/scan_cache.py:161]:
```suggestion
        `str`: The table as a string.
```

File: src/huggingface_hub/commands/scan_cache.py
Comment (Wauplin) [inline src/huggingface_hub/commands/scan_cache.py:158]:
```suggestion
        hf_cache_info ([`HFCacheInfo`]):
            The HFCacheInfo object to print.
        verbosity (`int`, *optional*):
            The verbosity level. Defaults to 0.
```

(nit) for docs styling

File: src/huggingface_hub/commands/scan_cache.py
Comment (Wauplin) [inline src/huggingface_hub/commands/scan_cache.py:137]:
```suggestion
    >>> hf_cache_info = scan_cache_dir()
    HFCacheInfo(...)
```

let's reduce this section to emphasize the table part rather than the `HFCacheInfo` part

File: src/huggingface_hub/commands/scan_cache.py
Comment (Wauplin) [inline src/huggingface_hub/commands/scan_cache.py:94]:
```suggestion

    Example:
```

Adding `example` section title will be recognized as such by the doc builder

File: src/huggingface_hub/commands/scan_cache.py
Comment (Wauplin) [inline src/huggingface_hub/commands/scan_cache.py:89]:
```suggestion
    Pass `verbosity=0` to get a table with a single row per repo, with columns
```

(nit)

File: src/huggingface_hub/commands/scan_cache.py
Comment (Wauplin) [inline src/huggingface_hub/commands/scan_cache.py:92]:
```suggestion
    Pass `verbosity=1` to get a table with a row per repo and revision (thus multiple rows can appear for a single repo), with columns
```

(same)

**Review by Wauplin — COMMENTED** [review]
Thanks for the changes @rsxdalv! I left minor comments mostly related to how the doc builder works. 

Regarding the `ko` documentation, could you add 

```
### scan_cache.get_table[[huggingface_hub.commands.scan_cache.get_table]]

[[autodoc]] huggingface_hub.commands.scan_cache.get_table
``` 

to it? (good catch, I forgot about it^^)

Other than that, we should be good to merge!

**Comment by rsxdalv** [comment]
Thanks, I accepted all the changes and added the ko/cache.md reference.

**Review by Wauplin — APPROVED** [review]
Great! Thank you @rsxdalv, I think we are good to go now :) Let's wait for the CI to complete and then I'll merge.

**Comment by Wauplin** [comment]
Hi @rsxdalv, I've had a look at the CI issues. It was quite annoying as the docs were failing because `get_table` was not a first-class citizen but we still wanted it in the docs. In the end, I went for another solution that should also suits you. See changes in https://github.com/huggingface/huggingface_hub/pull/2437/commits/d9584983fd77fefa1691184d3df9fdf40524bb0b if you are interested.

Code example is now:

```py
>>> from huggingface_hub.utils import scan_cache_dir
>>> hf_cache_info = scan_cache_dir()
HFCacheInfo(...)
>>> print(hf_cache_info.export_as_table())
REPO ID                                             REPO TYPE SIZE ON DISK NB FILES LAST_ACCESSED LAST_MODIFIED REFS LOCAL PATH
--------------------------------------------------- --------- ------------ -------- ------------- ------------- ---- --------------------------------------------------------------------------------------------------
roberta-base                                        model             2.7M        5 1 day ago     1 week ago    main C:\\Users\\admin\\.cache\\huggingface\\hub\\models--roberta-base
suno/bark                                           model             8.8K        1 1 week ago    1 week ago    main C:\\Users\\admin\\.cache\\huggingface\\hub\\models--suno--bark
t5-base                                             model           893.8M        4 4 days ago    7 months ago  main C:\\Users\\admin\\.cache\\huggingface\\hub\\models--t5-base
t5-large                                            model             3.0G        4 5 weeks ago   5 months ago  main C:\\Users\\admin\\.cache\\huggingface\\hub\\models--t5-large

```

Sorry about pushing changes to your branch, I hope that's fine for you. Once the CI is (finally) green, I'll merged it! :smile:

**Comment by rsxdalv** [comment]
Thank you for carrying this code change to it's inclusion! I am currently developing a module for managing the HF cache within my project, once it is more clear what works and what does not, I hope to make another PR.

One problem that I already know will need to be solved is distinguishing the different revisions from files. If a user sees that by deleting revision 3fd77fe... they will reclaim 10 GBs but actually those files are shared amongst 3 revisions that 'lock' them in, they will be confused. So now the only solution is to educate users about what are revisions and how 'results may warry' when deleting them.

**Comment by Wauplin** [comment]
Yes, I see what you mean. In the `delete-cache` command, each time an item is selected/unselected, we compute the actual size on disk that will be reclaimed. It's not as good as having it in the `scan-cache` command but it might help.

**Comment by rsxdalv** [comment]
Actually that does help, thanks for letting me know.

**Pull request merged**
